sys/ztimer/core.c

```diff
@@ -49,6 +48,45 @@ static inline uint32_t _min_u32(uint32_t a, uint32_t b)
 }
 #endif
 
+#if MODULE_ZTIMER_ONDEMAND
+bool _ztimer_acquire(ztimer_clock_t *clock)
+{
+    DEBUG("ztimer_acquire(): %p: %" PRIu16 " user(s)\n",
+          (void *)clock, clock->users + 1);
+
+    if (clock->users++ == 0) {
+        if (clock->ops->start) {
+            clock->ops->start(clock);
+        }
+
+        return true;
+    }
+
+    return false;
+}
+
+bool _ztimer_release(ztimer_clock_t *clock)
+{
+    assert(clock->users > 0);
```

Comment:
:+1: I have the feeling that countless hours of debugging will be significantly shorted down by this assert() :)

I wonder if it even would make sense to have `ztimer_acquire()` and `ztimer_release()` enforce matching pairs without `MODULE_ZTIMER_ONDEMAND` if `DEVELHELP` is set (or maybe in absence of `NDEBUG`). That would cost some memory, but could expose bugs that otherwise would only effect low power user cases.

Reply:
Good idea! So reference counting is done if `MODULE_ZTIMER_ONDEMAND` and/or `DEVELHELP` is set, right?